include/dlaf/multiplication/hermitian/api.h

```diff
@@ -10,16 +10,20 @@
 #pragma once
 
 #include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_ref.h>
 #include <dlaf/types.h>
 
 namespace dlaf::multiplication::internal {
+
+using matrix::internal::MatrixRef;
+
 template <Backend B, Device D, class T>
 struct Hermitian {
   static void call_LL(const T alpha, Matrix<const T, D>& mat_a, Matrix<const T, D>& mat_b, const T beta,
                       Matrix<T, D>& mat_c);
 
   static void call_LL(comm::CommunicatorGrid& grid, const T alpha, Matrix<const T, D>& mat_a,
-                      Matrix<const T, D>& mat_b, const T beta, Matrix<T, D>& mat_c);
+                      MatrixRef<const T, D>& mat_b, const T beta, Matrix<T, D>& mat_c);
```

Comment:
The hermitian multiplication is only used in the checks for the miniapps, an only use the distributed version. Since this is all in the `internal` namespace, I did not change everything to use `MatrixRef`. I agree the inconsistency is not great, I'll fix it.